dbms/src/Interpreters/ClusterProxy/DescribeStreamFactory.cpp

```diff
@@ -0,0 +1,56 @@
+#include <Interpreters/ClusterProxy/DescribeStreamFactory.h>
+#include <Interpreters/InterpreterDescribeQuery.h>
+#include <DataStreams/MaterializingBlockInputStream.h>
+#include <DataStreams/BlockExtraInfoInputStream.h>
+#include <DataStreams/RemoteBlockInputStream.h>
+
+namespace DB
+{
+
+namespace
+{
+
+BlockExtraInfo toBlockExtraInfo(const Cluster::Address & address)
```

Comment:
TODO: We should totally get rid of `BlockExtraInfo`, because the same functionality is possible with WITH section in SELECT query.

`BlockExtraInfo` is harmful, because it cost "61 matches in 14 files" for almost no benefit.